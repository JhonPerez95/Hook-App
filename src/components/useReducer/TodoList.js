import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group">
      {todos.map((item, i) => (
        <TodoListItem
          key={item.id}
          index={i}
          item={item}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
        />
      ))}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoList;
