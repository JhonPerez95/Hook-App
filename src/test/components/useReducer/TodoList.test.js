import React from 'react';
import { shallow } from 'enzyme';

import TodoList from '../../../components/useReducer/TodoList';
import { demoTodos } from './demoTodos';

describe('Pruebas al componente <TodoList/>', () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );
  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe renderizar la misma cantidad de ToDos', () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);

    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(
      expect.any(Function)
    );
    expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual(
      expect.any(Function)
    );
  });
});
