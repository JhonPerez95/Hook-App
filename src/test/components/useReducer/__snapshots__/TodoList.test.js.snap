// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <TodoList/> Debe mostrarse correctamente 1`] = `
<ul
  className="list-group"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    item={
      Object {
        "description": " Learning React",
        "done": false,
        "id": 1,
      }
    }
    key="1"
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    item={
      Object {
        "description": " Learning Nest",
        "done": false,
        "id": 2,
      }
    }
    key="2"
  />
</ul>
`;
